### PR TITLE
Removing organization representative type from Find a Representative

### PR DESCRIPTION
**Pull request by cosu419**
## Summary
We are removing Claims Agents and Organizations from the Find a Representative search results. As such, the backend needs to stop supporting these two rep types.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/71816

## What areas of the site does it impact?
Find a Representative app which has not yet gone live

### CODE REVIEW

File: modules/veteran/app/models/veteran/service/organization.rb
Comment (jvcAdHoc) [inline modules/veteran/app/models/veteran/service/organization.rb:1]: The model shouldn't be deleted

File: modules/veteran/app/sidekiq/rep_org_addresses/update_addresses.rb [hunk 1]
```diff
@@ -66,10 +66,7 @@ def address_valid?(response)
     # @param api_response [Hash] The response from the address validation service.
     def update_address_record(data, api_response)
       record =
-        if data['type'] == 'representative'
           Veteran::Service::Representative.find_by(representative_id: data['id'])
-        elsif data['type'] == 'organization'
-          Veteran::Service::Organization.find_by(poa: data['id'])
         end
```
Comment (holdenhinkle) [inline modules/veteran/app/sidekiq/rep_org_addresses/update_addresses.rb:70]: Remove this `end`

File: modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb [hunk 2]
```diff
@@ -81,65 +78,54 @@ def create_json_data(row, sheet_name, column_map)
...
-        build_data_for_vsos(common_data, row, column_map).to_json
+      else
+        log_error("Unexpected sheet encountered: #{sheet_name}")
+        {}.to_json
       end
     rescue => e
       log_error("Error transforming data to JSON for #{sheet_name}: #{e.message}")
     end
 
     def build_common_data(row, sheet_name, column_map)
```
Comment (holdenhinkle) [inline modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb:89]: remove `sheet_name` from params

File: modules/veteran/app/controllers/veteran/v0/accredited_representatives_controller.rb [hunk 3]
```diff
@@ -15,9 +15,8 @@ class AccreditedRepresentativesController < ApplicationController
       DEFAULT_PER_PAGE = 10
       DEFAULT_SORT = 'distance_asc'
 
-      PERMITTED_TYPES = %w[attorney claim_agents organization].freeze
+      PERMITTED_TYPES = %w[attorney].freeze
```
Comment (holdenhinkle) [inline modules/veteran/app/controllers/veteran/v0/accredited_representatives_controller.rb:18]: This should just be 

`PERMITTED_TYPES = 'attorney'`

for now.

**Review by holdenhinkle — CHANGES_REQUESTED** [review]
Jon and I made a few comments.

Be sure to run `rubocop your_filename.rb` on each file that has linting and security issues.

File: modules/veteran/app/models/veteran/service/organization.rb
Comment (cosu419) [reply modules/veteran/app/models/veteran/service/organization.rb:1]: Restored

File: modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb (hunk 2 above)
Comment (cosu419) [reply modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb]: Done

File: modules/veteran/app/sidekiq/rep_org_addresses/update_addresses.rb (hunk 1 above)
Comment (cosu419) [reply modules/veteran/app/sidekiq/rep_org_addresses/update_addresses.rb]: Done

File: modules/veteran/app/controllers/veteran/v0/accredited_representatives_controller.rb [hunk 4]
```diff
@@ -32,24 +31,12 @@ def index
...
-        @organization_search ||= search_params[:type] == 'organization'
-      end
-
       def model_klass
-        @model_klass ||= if organization_search?
-                           'Veteran::Service::Organization'.constantize
-                         else
-                           'Veteran::Service::Representative'.constantize
-                         end
+        @model_klass ||= 'Veteran::Service::Representative'.constantize
```
Comment (jvcAdHoc) [inline modules/veteran/app/controllers/veteran/v0/accredited_representatives_controller.rb:35]: Some of the "klass" stuff isn't needed anymore now that organizations aren't searchable

File: modules/veteran/spec/models/veteran/service/organization_spec.rb
Comment (jvcAdHoc) [inline modules/veteran/spec/models/veteran/service/organization_spec.rb:1]: This spec shouldn't get deleted

File: modules/veteran/app/sidekiq/veteran/vso_reloader.rb
Comment (jvcAdHoc) [inline modules/veteran/app/sidekiq/veteran/vso_reloader.rb:1]: I don't think we want to make any changes to the nightly ogc ingestion. The proposed changes would stop new veteran service officers and organizations from being created.

File: modules/veteran/spec/models/veteran/service/organization_spec.rb
Comment (cosu419) [reply modules/veteran/spec/models/veteran/service/organization_spec.rb:1]: Reverted

**Review by holdenhinkle — CHANGES_REQUESTED** [review]

File: modules/veteran/app/sidekiq/veteran/vso_reloader.rb
Comment (cosu419) [reply modules/veteran/app/sidekiq/veteran/vso_reloader.rb:1]: Reverted

**Review by holdenhinkle — DISMISSED** [review]
LGTM

**Review by holdenhinkle — DISMISSED** [review]
LGTM

File: modules/veteran/spec/requests/v0/accredited_representatives_spec.rb [hunk 5]
```diff
@@ -126,112 +126,6 @@
...
-        expect(parsed_response['meta']['pagination']['total_entries']).to eq(0)
-      end
-    end
-
-    it 'returns ok for a successful request' do
-      get '/services/veteran/v0/accredited_representatives',
-          params: { type: 'organization', lat: 40.7128, long: -74.0060 }
-
-      expect(response).to have_http_status(:ok)
-    end
```
Comment (jvcAdHoc) [inline modules/veteran/spec/requests/v0/accredited_representatives_spec.rb:146]: These two specs are still valuable as they describe behavior that is still in the controller. Just need to change `organization` to `attorney`.

Comment (holdenhinkle) [reply modules/veteran/spec/requests/v0/accredited_representatives_spec.rb:146]: Okay, thanks.

Comment (holdenhinkle) [reply modules/veteran/spec/requests/v0/accredited_representatives_spec.rb]: Thanks, Jon. I added them back.

**Review by holdenhinkle — APPROVED** [review]
LGTM

**Pull request merged**